### DISCUSSION TITLE
Inputfields don't work inside of a nested dialog

### DISCUSSION THREAD

**Discussion by RamadaniLeotrim** [microsoft/fluentui-blazor#1984] [Q&A] [answered]
Hello everyone,

I want to put a EditForm inside a Dialog which is opened from another Dialog. I noticed that you can not type into a textfield. I tried it with other input components but nothing seems to work.

**Comment by vnbaaij** [comment]
That is probably because a Dialog uses an Overlay to shield off the background. If you have two, they will overlap.

We have not claimed (nor tested) nesting dialogs to work...

**Reply by RamadaniLeotrim** [reply]
Hi I tried it again in hope of finding a workaround, but I just noticed that there is an error in the web-components.js file. 

<img width="254" alt="Screenshot 2024-05-07 081407" src="https://github.com/microsoft/fluentui-blazor/assets/81311415/742e0388-3ac0-4697-b405-86de1b45dd5c">

<img width="209" alt="Screenshot 2024-05-07 081449" src="https://github.com/microsoft/fluentui-blazor/assets/81311415/8f16359f-cd0e-4786-9126-e680dcf56e4f">

**Comment by RamadaniLeotrim** [comment]
Thank you for your reply.

Are nested Dialog going to work in the future or is it not planned?

**Reply by dvoituron** [reply, marked as answer]
In FluentUI, the use of nested dialogs is generally discouraged. According to the FluentUI documentation, selecting an action in a dialog should move users forward in their workflow. [Having multiple dialogs can distract from the original context and make it difficult for users, especially those with cognitive disabilities or who use magnified views, to interact with the content](https://develop.fluentui.dev/components/web/react/dialog/usage).

For best usability and to ensure accessibility, it’s recommended to avoid nesting dialogs. [Instead, dialogs should help people take action by keeping options simple and focused on the main goal](https://develop.fluentui.dev/components/web/react/dialog/usage). If you need to present additional content or actions, consider using alternative UI elements that complement the workflow without the need for multiple overlapping dialogs.

We don't forbid their use, but we haven't had time to validate everything.

**Reply by RamadaniLeotrim** [reply]
Ok, I understand. Thank you for your response.

**Comment by MrKelvic** [comment]
Set the TrapFocus parameter to false

**Reply by DomenicoFormoso** [reply]
it work for me

**Comment by biralpucar** [comment]
If you don’t have a requirement to keep the dialog open below, using dialog.Hide() before waiting for the dialog result and then dialog.Show() afterwards resolves the trap focus issue.
